### PR TITLE
Update DG with Test Cases for Manual Testing Part 2/2

### PR DESCRIPTION
**Pull request by peaol**
==Ready to merge==

You can view the DG [here](https://github.com/peaol/tp/blob/branch-DG/docs/DeveloperGuide.md)

Added in test cases for the following sections under `Appendix: Instructions for manual testing`:

- Students Page Commands
- Session Page Commands
- Attendance Page Commands

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -714,20 +714,158 @@ These commands should be tested on the Group Page.
 ### Students Page Commands
 These commands should be tested on the Students Page.
 
+#### Adding a student
+
+1. Adding a student while the student list is empty or the student does not exist in the course list.
+
+   1. Test case: ` add student n/John Doe p/98765432 e/johnd@example.com sid/A1234567X`<br>
```
Comment (tyx021) [inline docs/DeveloperGuide.md:721]:
```suggestion
   1. Test case: `add student n/John Doe p/98765432 e/johnd@example.com sid/A1234567X`<br>
```

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -714,20 +714,158 @@ These commands should be tested on the Group Page.
...
+      Expected: New student is added into the list. Details of the added student shown in the log box.
+
+   2. Other incorrect `add student` commands to try: `add`, `add student` (where the name, phone number, email or student ID is empty)<br>
```
Comment (tyx021) [inline docs/DeveloperGuide.md:724]:
```suggestion
   2. Other incorrect `add student` commands to try: `add`, `add student` (where the name, phone number, email or student ID is not specified)<br>
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -714,20 +714,158 @@ These commands should be tested on the Group Page.
...
+      Expected: No student is added. Error details shown in the log box.
+
+2. Adding a student whose student ID already exists in the list.
+
+    1. Test case: `add student n/John p/98765432 e/johnd@example.com sid/A1234567X`, where another student with student ID A1234567X already exists in the student list<br>
+       Expected: No course is added. Error details shown in the log box.
```
Comment (tyx021) [inline docs/DeveloperGuide.md:730]:
```suggestion
       Expected: No student is added. Error details shown in the log box.
```

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -714,20 +714,158 @@ These commands should be tested on the Group Page.
...
+   3. Other incorrect `remark` commands to try: `remark`, `remark x`, (where x is larger than the student lit)
+   Expected: Displays error message on the log box.
+2. Adding a remark to a student that already has a remark.
+   1. Test case: `remark 1 Plays football` 
+      Expected: Replaces the previous remark that the first student has with `Plays football`. Details of the new remark displayed in the log box.
+   2. Test case: `remark 0 Loves sci-fi movies`
+     Expected: Displays error message on the log box.
+   3. Other incorrect `remark` commands to try: `remark`, `remark x`, (where x is larger than the student lit)
+      Expected: Displays error message on the log box.
+
```
Comment (tyx021) [inline docs/DeveloperGuide.md:856]: Maybe can add cases of deleting remark, it was mentioned in the title but not listed down

**Review by tyx021 — COMMENTED** [review]
Good job 👍

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -714,20 +714,158 @@ These commands should be tested on the Group Page.
...
+1. Adding a student while the student list is empty or the student does not exist in the course list.
+
+   1. Test case: ` add student n/John Doe p/98765432 e/johnd@example.com sid/A1234567X`<br>
+      Expected: New student is added into the list. Details of the added student shown in the log box.
+
+   2. Other incorrect `add student` commands to try: `add`, `add student` (where the name, phone number, email or student ID is empty)<br>
+      Expected: No student is added. Error details shown in the log box.
+
+2. Adding a student whose student ID already exists in the list.
```
Comment (IsabelChong) [inline docs/DeveloperGuide.md:727]:
```suggestion
2. Adding a student whose student ID already exists in the list.<br>
```

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -714,20 +714,158 @@ These commands should be tested on the Group Page.
...
+    1. Test case: `delete student 1`<br>
+    Expected: No student is deleted. Error details shown in the log box.
+
+#### Copying a student's email
+1. Copying a student's email to your clipboard, while there is more than one student on the Students Page.
+    1. Test case: `copy 1`<br>
+    Expected: First student's email is copied to your clipboard. Successful command message shown in the log box.
+   2. Test case: `copy 0`<br>
+   Expected: No student email is copied. Error details shown in the log box.
+   3. Other incorrect `copy` commands to try: `copy`, `copy x`, (where x is larger than the student list)
```
Comment (IsabelChong) [inline docs/DeveloperGuide.md:827]:
```suggestion
   3. Other incorrect `copy` commands to try: `copy`, `copy x`, (where x is larger than the student list)<br>
```

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -714,20 +714,158 @@ These commands should be tested on the Group Page.
...
+    1. Test case: `sort id`
+   Expected: Student list sorted by student ID.
+
+#### Adding or deleting a remark
+1. Adding a remark to a student that does not have an existing remark.
+    1. Test case: `remark 1 Loves photography`
+   Expected: Adds a remark `Loves photography` to the first student on the student list. Details of the new remark displayed in the log box.
+   2. Test case: `remark 0 Loves sci-fi movies`
+   Expected: Displays error message on the log box.
+   3. Other incorrect `remark` commands to try: `remark`, `remark x`, (where x is larger than the student lit)
```
Comment (IsabelChong) [inline docs/DeveloperGuide.md:847]:
```suggestion
   3. Other incorrect `remark` commands to try: `remark`, `remark x`, (where x is larger than the student list)<br>
```

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -714,20 +714,158 @@ These commands should be tested on the Group Page.
...
+   Expected: No student email is copied. Error details shown in the log box.
+   3. Other incorrect `copy` commands to try: `copy`, `copy x`, (where x is larger than the student list)
+   Expected: No student email is copied. Error details shown in the log box.
+2. Copying a student's email while the student list is empty on the Students Page.
+    1. Test case: `copy 1`<br>
+    Expected: Nothing is copied to your clipboard. Error details shown in the log box.
+
+#### Sorting students by their name or student ID
+1. Sorting student list by their name
+    1. Test case: `sort name`
```
Comment (IsabelChong) [inline docs/DeveloperGuide.md:835]:
```suggestion
    1. Test case: `sort name`<br>
```

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -714,20 +714,158 @@ These commands should be tested on the Group Page.
...
+   Expected: Student list sorted by name in alphabetical order.
+2. Sorting student list by their student IDs
+    1. Test case: `sort id`
```
Comment (IsabelChong) [inline docs/DeveloperGuide.md:838]:
```suggestion
    1. Test case: `sort id`<br>
```

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -714,20 +714,158 @@ These commands should be tested on the Group Page.
...
+   Expected: Student list sorted by student ID.
+
+#### Adding or deleting a remark
+1. Adding a remark to a student that does not have an existing remark.
+    1. Test case: `remark 1 Loves photography`
```
Comment (IsabelChong) [inline docs/DeveloperGuide.md:843]:
```suggestion
    1. Test case: `remark 1 Loves photography`<br>
```

File: docs/DeveloperGuide.md [hunk 11]
```diff
@@ -714,20 +714,158 @@ These commands should be tested on the Group Page.
...
+   Expected: Adds a remark `Loves photography` to the first student on the student list. Details of the new remark displayed in the log box.
+   2. Test case: `remark 0 Loves sci-fi movies`
```
Comment (IsabelChong) [inline docs/DeveloperGuide.md:845]:
```suggestion
   2. Test case: `remark 0 Loves sci-fi movies`<br>
```

File: docs/DeveloperGuide.md [hunk 12]
```diff
@@ -714,20 +714,158 @@ These commands should be tested on the Group Page.
...
+   Expected: Displays error message on the log box.
+   3. Other incorrect `remark` commands to try: `remark`, `remark x`, (where x is larger than the student lit)
+   Expected: Displays error message on the log box.
+2. Adding a remark to a student that already has a remark.
+   1. Test case: `remark 1 Plays football`
```
Comment (IsabelChong) [inline docs/DeveloperGuide.md:850]:
```suggestion
   1. Test case: `remark 1 Plays football` <br>
```

File: docs/DeveloperGuide.md [hunk 13]
```diff
@@ -714,20 +714,158 @@ These commands should be tested on the Group Page.
...
+    1. Test case: `remark 1 Loves photography`
+   Expected: Adds a remark `Loves photography` to the first student on the student list. Details of the new remark displayed in the log box.
+   2. Test case: `remark 0 Loves sci-fi movies`
+   Expected: Displays error message on the log box.
+   3. Other incorrect `remark` commands to try: `remark`, `remark x`, (where x is larger than the student lit)
+   Expected: Displays error message on the log box.
+2. Adding a remark to a student that already has a remark.
+   1. Test case: `remark 1 Plays football` 
+      Expected: Replaces the previous remark that the first student has with `Plays football`. Details of the new remark displayed in the log box.
+   2. Test case: `remark 0 Loves sci-fi movies`
```
Comment (IsabelChong) [inline docs/DeveloperGuide.md:852]:
```suggestion
   2. Test case: `remark 0 Loves sci-fi movies`<br>
```

File: docs/DeveloperGuide.md [hunk 14]
```diff
@@ -714,20 +714,158 @@ These commands should be tested on the Group Page.
...
+     Expected: Displays error message on the log box.
+   3. Other incorrect `remark` commands to try: `remark`, `remark x`, (where x is larger than the student lit)
```
Comment (IsabelChong) [inline docs/DeveloperGuide.md:854]:
```suggestion
   3. Other incorrect `remark` commands to try: `remark`, `remark x`, (where x is larger than the student list)<br>
```

File: docs/DeveloperGuide.md [hunk 15]
```diff
@@ -736,13 +874,139 @@ These commands should be tested on the Students Page.
 ### Session Page Commands
 These commands should be tested on the Session Page.
 
+#### Adding a session
+1. Adding a session while the session list is empty, or the session does not exist in the session list.
+
+    1. Test case: `add session Tutorial1`<br>
+       Expected: New session is added into the list. Details of the added task shown in the log box.
```
Comment (IsabelChong) [inline docs/DeveloperGuide.md:881]:
```suggestion
       Expected: New session is added into the list. Details of the added session shown in the log box.
```

File: docs/DeveloperGuide.md [hunk 16]
```diff
@@ -736,13 +874,139 @@ These commands should be tested on the Students Page.
...
+
+    1. Incorrect `delete session` commands to try: `delete`, `delete session x` (where x is any number)<br>
+       Expected: No session is deleted. Error details shown in the log box.
+
+#### Editing a session
+
+1. Editing a session which exists in the Session list.
+
+   1. Test case: `edit session 1 Tutorial4`<br>
+      Expected: First session is edited to the new task name. Details of the edited session shown in the log box.
```
Comment (IsabelChong) [inline docs/DeveloperGuide.md:917]:
```suggestion
      Expected: First session is edited to the new session name. Details of the edited session shown in the log box.
```

File: docs/DeveloperGuide.md [hunk 17]
```diff
@@ -736,13 +874,139 @@ These commands should be tested on the Students Page.
...
+#### Marking selected student(s) as present
+
+1. Marking selected student as present, with at least one student in the list.
+
+   1. Test case: `mark 1`<br>
+      Expected: First student is marked as present. Details of the student marked as present shown in the log box.
+
+   2. Other incorrect `mark` commands to try: `mark`, `mark x` (where x is larger than the student list size)
+      , `mark -1` or `mark 101` (where x is a valid index from the student list) <br>
+      Expected: No student is marked as present. Error details shown in the log box.
```
Comment (IsabelChong) [inline docs/DeveloperGuide.md:976]:
```suggestion
   2. Other incorrect `mark` commands to try: `mark`, `mark x` (where x is larger than the student list size, `mark -1` <br>
      Expected: No student is marked as present. Error details shown in the log box.
```


Reason: `mark 101` is a correct command if index 101 exists for `at least one student`.

File: docs/DeveloperGuide.md [hunk 18]
```diff
@@ -736,13 +874,139 @@ These commands should be tested on the Students Page.
...
+   
+#### Marking selected student(s) as absent
+1. Marking selected student as absent, with at least one student in the list.
+
+    1. Test case: `unmark 1`<br>
+       Expected: First student is marked as absent. Details of the student marked as absent shown in the log box.
+
+    2. Other incorrect `unmark` commands to try: `unmark`, `unmark x` (where x is larger than the student list size)
+       , `unmark -1` or `unmark 101` (where x is a valid index from the student list) <br>
+       Expected: No student is marked as absent. Error details shown in the log box.
```
Comment (IsabelChong) [inline docs/DeveloperGuide.md:994]:
```suggestion
    2. Other incorrect `unmark` commands to try: `unmark`, `unmark x` (where x is larger than the student list size), `unmark -1`<br>
       Expected: No student is marked as absent. Error details shown in the log box.
```

Reason: `unmark 101` is a correct command if index 101 exists for at least one student.

**Review by IsabelChong — COMMENTED** [review]
Good job! Just some minor fixes 👍

**Pull request merged**